coderd/rbac/authz.go

```diff
@@ -194,5 +257,27 @@ func (a RegoAuthorizer) PrepareByRoleName(ctx context.Context, subjectID string,
 		return nil, err
 	}
 
-	return a.Prepare(ctx, subjectID, roles, scopeRole, groups, action, objectType)
+	prepared, err := a.Prepare(ctx, subjectID, roles, scopeRole, groups, action, objectType)
+	if err != nil {
+		return nil, err
+	}
+
+	span.AddEvent("prepared",
+		trace.WithAttributes(attribute.Int("num_queries", len(prepared.preparedQueries))),
+		trace.WithAttributes(attribute.Bool("always_true", prepared.alwaysTrue)),
+	)
```

Comment:
I think instead of adding an event to attach these attributes it'd make more sense to just attach them to the parent span above, or just give `(RegoAuthorizer).Prepare` it's own span?

Reply:
I think it might just be better to call `tracing.StartSpan()` inside prepare then? That way you have the timing for that function specifically and can add attributes to it

Reply:
The attributes are the return though. So even if I put it in, I cannot put those attributes at the start of the call.

Reply:
`tracing.StartSpan` returns a span you can set the attributes on

Reply:
Oh TIL I can add attributes after a span was stared. I thought I read that you could not. I'll just add to parent span :+1: